packages/wdio-browserstack-service/src/launcher.ts

```diff
@@ -44,11 +50,13 @@ export default class BrowserstackLauncherService implements Services.ServiceInst
         if (Array.isArray(capabilities)) {
             capabilities.forEach((capability: Capabilities.DesiredCapabilities) => {
                 if (!capability['bstack:options']) {
-                    const extensionCaps = Object.keys(capability).filter((cap) => cap.includes(':'))
-                    if (extensionCaps.length) {
-                        capability['bstack:options'] = { wdioService: bstackServiceVersion }
-                    } else if (shouldAddServiceVersion(this._config, this._options.testObservability)) {
-                        capability['browserstack.wdioService'] = bstackServiceVersion
+                    if (isBStackSession(this._config)) {
```

Comment:
I created https://github.com/webdriverio/webdriverio/issues/10039 .. we could simplify the implementation by tackling that first, up to you.

Reply:
@christian-bromann the nested if condition is for checking whether we need to set `wdioService` version or not. After if condition there is code to set the `buildIdentifier` and `buildName` which we want to set even if it is not a browserstack session. So moving this to constructor won't help.

Reply:
> After if condition there is code to set the `buildIdentifier` and `buildName` which we want to set even if it is not a browserstack session.

Can you help me understand why this service wants to set these properties even if the session is not run on BS?

Reply:
We have a product called observability, which shows insights of tests. This is independent of the platform in which the tests are run. So tests can run browserstack, local or somewhere else, we want the collect tests data. So that's why we are setting `buildIdentifier` and `buildName` even if it is not a browserstack session.

Reply:
Gotcha, makes sense. Mind adding a comment for outside folks who are not familiar with this?

Reply:
Sure, have added https://github.com/webdriverio/webdriverio/pull/10037/commits/9c0e75c2c0739a7756b9f4a171c66d27988b3a1e

Reply:
So just to understand this correctly. Users can use the BS service even though they run their tests on a local chrome browser without having BS involved at all?

Reply:
Yes @christian-bromann, it's used for reporting like https://github.com/adamgruber/mochawesome